src/test/rustdoc-ui/check-doc-attrs.stderr

```diff
@@ -0,0 +1,52 @@
+warning: this attribute can only be applied to a `use` item
+  --> $DIR/check-doc-attrs.rs:1:8
+   |
+LL | #![doc(inline)]
+   |        ^^^^^^
+   |
+   = note: read https://doc.rust-lang.org/nightly/rustdoc/the-doc-attribute.html#docno_inlinedocinline for more information
+
+error: conflicting doc inlining attributes
+  --> $DIR/check-doc-attrs.rs:4:7
+   |
+LL | #[doc(inline)]
+   |       ^^^^^^ this attribute...
+LL | #[doc(no_inline)]
+   |       ^^^^^^^^^ ...conflicts with this attribute
+   |
+   = help: remove one of the conflicting attributes
+
+warning: this attribute can only be applied to a `use` item
+  --> $DIR/check-doc-attrs.rs:13:7
+   |
+LL | #[doc(no_inline)]
+   |       ^^^^^^^^^
+   |
+   = note: read https://doc.rust-lang.org/nightly/rustdoc/the-doc-attribute.html#docno_inlinedocinline for more information
+
+warning: this attribute can only be applied at the crate level
+  --> $DIR/check-doc-attrs.rs:15:7
+   |
+LL | #[doc(html_no_source)]
+   |       ^^^^^^^^^^^^^^
+   |
+   = note: read https://doc.rust-lang.org/nightly/rustdoc/the-doc-attribute.html#at-the-crate-level for more information
+
+warning: this attribute can only be applied to a `use` item
+  --> $DIR/check-doc-attrs.rs:19:7
+   |
+LL | #[doc(no_inline)]
+   |       ^^^^^^^^^
+   |
+   = note: read https://doc.rust-lang.org/nightly/rustdoc/the-doc-attribute.html#docno_inlinedocinline for more information
+
+warning: this attribute can only be applied at the crate level
+  --> $DIR/check-doc-attrs.rs:21:7
+   |
+LL | #[doc(html_no_source)]
```

Comment:
Maybe suggest `#![doc(html_no_source)]` instead? https://doc.rust-lang.org/nightly/nightly-rustc/rustc_errors/struct.Diagnostic.html#method.span_suggestion

Reply:
Good idea! But only if the enclosing scope is the crate, I guess

Reply:
This should only happen for items in the current crate, right? Or at least, if not that's a bug - we shouldn't be linting other crates.

Reply:
>  But only if the enclosing scope is the crate

Oh I misread this - yes, ideally this would only warn if this is an outer attribute on an item with the crate root as the parent. You can find that out with `tcx.parent(item).index == CRATE_DEF_INDEX`.